include/extractor/guidance/intersection_handler.hpp

```diff
@@ -508,15 +508,15 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
             // even reverse the direction. Since we don't want to compute actual turns but simply
             // try to find whether there is a turn going to the opposite direction of our obvious
             // turn, this should be alright.
-            NodeID new_node;
-            const auto previous_intersection = [&]() {
-                EdgeID turn_edge;
-                std::tie(new_node, turn_edge) = intersection_generator.SkipDegreeTwoNodes(
+            const auto previous_intersection = [&]() -> IntersectionView {
+                const auto parameters = intersection_generator.SkipDegreeTwoNodes(
```

Comment:
Hm. I haven't thought about it much, yet. But I'd say this is a different task, not related to this refactoring quest. We have two tickets regarding the performance already. I've added this as an additional task to https://github.com/Project-OSRM/osrm-backend/issues/3473.